tests/e2e/Services/Projects/ProjectsDevKeys.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+namespace Tests\E2E\Services\Projects;
+
+use Tests\E2E\Client;
+use Utopia\Database\DateTime;
+
+trait ProjectsDevKeys
+{
+    /**
+     * @depends testCreateProject
+     * @group devKeys
+     */
+    public function testCreateProjectDevKey($data): array
+    {
+        $id = $data['projectId'] ?? '';
+
+        $response = $this->client->call(Client::METHOD_POST, '/projects/' . $id . '/development-keys', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'name' => 'Key Test',
+            'expire' => DateTime::addSeconds(new \DateTime(), 36000)
+        ]);
+
+        $this->assertEquals(201, $response['headers']['status-code']);
+        $this->assertNotEmpty($response['body']['$id']);
+        $this->assertEquals('Key Test', $response['body']['name']);
+        $this->assertNotEmpty($response['body']['secret']);
+        $this->assertArrayHasKey('sdks', $response['body']);
+        $this->assertEmpty($response['body']['sdks']);
+        $this->assertArrayHasKey('accessedAt', $response['body']);
+        $this->assertEmpty($response['body']['accessedAt']);
+
+        /** TEST expiry date is required */
```

Comment:
lets also test:
- Cant make 2 keys with same ID
- Cant make key with expiry in past

Reply:
What do you mean 2 keys with same ID? keys don't support custom ID. And for expire, we don't do that test in the Project API keys as well, so I'm doing what we are doing in the API keys.

Reply:
For expire, I would vote to do it here, as well as project API keys.
Let's please add TODO in both places, and tell Christy about those todos. These can be great tasks for interns.